net/url_request/url_fetcher.h

```diff
@@ -311,6 +311,14 @@ class NET_EXPORT URLFetcher {
   virtual void SaveResponseToTemporaryFile(
       scoped_refptr<base::SequencedTaskRunner> file_task_runner) = 0;
 
+#if defined(IN_MEMORY_UPDATES)
+  // By default, the response is saved in a string without preallocation. Call
+  // this method when the response is expected to be large and the capacity
+  // should be reserved to the response's content size upfront to reduce
+  // fragmentation.
```

Comment:
Did we measure or benchmark existing implementation that fragmentation + reallocation is a real concern here ? 

Also, there's a `SaveResponseWithWriter()`  below which allows providing a custom response writer class without modifying the API here. Couldn't we simply make a URLFetcherResponseWriter that pre-allocates the string ?

Reply:
35 MB seems very minor, when just media pipeline itself takes 150-200 MB. If we are doing updating _while media isnt playing_ then the 35 megabyte number is lost in the noise. It seems like a micro-optimization that doesn't look at the big picture. 

We are pumping through 60-70 **megabits** per second playing 4K and even 4K video through XHR / Fetch / standard net fetcher codepaths. Downloading a binary should _not_ be enough of a justification to amend a proven, stable API.

Reply:
Updates are *not* guaranteed to happen while media isn't playing. Updates happen in the background at scheduled intervals and we have no logic to prevent them when other things (like media) are happening, or any plans that I'm aware of to do this.

Reply:
Resolved in internal discussion. Lets follow up with looking at what would it take to disable updates while media is playing